src/ec/eddsa.rs

```diff
@@ -22,18 +22,50 @@ pub struct EdDSAParameters;
 
 /// An Ed25519 key pair, for signing.
 pub struct Ed25519KeyPair {
-    private_public: [u8; KEY_PAIR_LEN],
+    private_key: [u8; SEED_LEN],
+
+    // As described in RFC 8032 Section 5.1.6, Step 1, we compute and cache
+    // these values when generating key pairs, since they are used in every
+    // signature. They are defined by:
+    //
+    //     SHA-512(private_key) = private_scalar || private_prefix
+    //
```

Comment:
I think `private_scalar` should be masked before being stored here so that it is a true scalar.

I think we should either note that `private_scalar` is “the secret scalar *s*` in the RFC and that `private_prefix` is *prefix* from the RFC, or we should just name these `s` and `prefix` respectively and note that the names come from the RFC.

I'd rather not document how they are calculated since that will be evident from the code below.